### PR TITLE
939 old find params

### PR DESCRIPTION
**Pull request by goodviber**
### Context
We've changed Find query params in the migrated version to use the same search params as the CourseSearchService to make it easier to maintain.

We should try to handle URLs with the old query params as well.
What needs to be done

Figure out a way to check an incoming url for old find query params and transform/redirect the request to the new query params so the user still gets to the same place

Success

Given an old find url such as /results?age_group=secondary&l=2&subject_codes%5B%5D=W1&subject_codes%5B%5D=C1&subject_codes%5B%5D=08 the user should be taken to the new find equivalent
Tech notes

Could use a routing constraint to check for the old url within a redirect block in the routes with some additional logic to transform the url
We'll probably only need to check urls beginning with the /results* prefix.
I think we can only check one way eg first land on the page, once the user starts modifying their search they'll be using the new params anyway
Keep it simple this is just to enhance and preserve UX

### Changes proposed in this pull request
Creation of a params mapping service to be called in the results :index
### Guidance to review
run the new results page in the browser with params from old find
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by goodviber** [comment]
[Handle old find query params](https://trello.com/c/coZIGMe8/939-handle-old-find-query-params)

**Comment by darokel** [comment]
@goodviber Could you rebase so we could get the constraint in here to see the review app?

File: .github/actions/smoke-test/action.yml [hunk 1]
```diff
@@ -40,14 +40,14 @@ runs:
       shell: bash
       run: RAILS_ENV=${{ inputs.environment }} ./bin/bundle exec rspec spec/smoke --format documentation
 
-    - name: 'Notify #twd_findpub_tech on failure'
```
Comment (defong) [inline .github/actions/smoke-test/action.yml:43]: Umm can we revert these changes please

File: app/services/find/match_old_params.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      "Other" => "pgce pgde",
+      "PgdePgceWithQts" => "pgce_with_qts",
+      "QtsOnly" => "qts",
+    }.freeze
+
+    def initialize(request_params)
+      @request_params = request_params
+    end
+
+    def call
```
Comment (defong) [inline app/services/find/match_old_params.rb:30]: Can we break this method to smaller private methods, akin to the way you have tested it

@darokel I think we need to revisit https://github.com/DFE-Digital/publish-teacher-training/blob/main/config/rubocop/metrics.yml

It seems we induced needless tech debt.

Comment (goodviber) [reply app/services/find/match_old_params.rb:30]: Done

File: .github/actions/smoke-test/action.yml (hunk 1 above)
Comment (goodviber) [reply .github/actions/smoke-test/action.yml]: vimmed

File: app/services/find/match_old_params.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  private
+
+    def sortby_and_funding
+      @request_params["sortby"] = "distance" if @request_params["sortby"] == "2"
+      @request_params["funding"] = "salary" if @request_params["funding"] == "8"
+    end
+
+    def qualification_filters
+      if @request_params["qualifications"]
+        @request_params["qualification"] = @request_params.delete("qualifications")
```
Comment (defong) [inline app/services/find/match_old_params.rb:48]: :woozy_face:

File: app/views/find/result_filters/_hidden_fields.html.erb [hunk 4]
```diff
@@ -9,3 +9,9 @@
     <%= form.hidden_field(:subjects, multiple: true, value: subject_code) %>
   <% end %>
 <% end %>
+<!-- to preserve old find subject codes -->
```
Comment (defong) [inline app/views/find/result_filters/_hidden_fields.html.erb:12]:
```suggestion
<!-- NOTE: to preserve old find subject codes -->
```

File: app/views/layouts/find_layout.html.erb [hunk 5]
```diff
@@ -53,7 +53,6 @@
         ),
       ) %>
     <% end %>
-
```
Comment (defong) [inline app/views/layouts/find_layout.html.erb:56]: Can we have it back please.

File: spec/services/find/match_old_params_spec.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
+require "rails_helper"
+# just a comment
```
Comment (defong) [inline spec/services/find/match_old_params_spec.rb:2]:
```suggestion

```

File: app/services/find/match_old_params.rb (hunk 3 above)
Comment (goodviber) [reply app/services/find/match_old_params.rb:48]: I think that is correct?

File: app/views/layouts/find_layout.html.erb (hunk 5 above)
Comment (goodviber) [reply app/views/layouts/find_layout.html.erb:56]: Done

File: app/services/find/match_old_params.rb (hunk 3 above)
Comment (defong) [reply app/services/find/match_old_params.rb:48]: I not sure what you are intending....

image

Comment (goodviber) [reply app/services/find/match_old_params.rb:48]: x = Hash.new
x[:qualifications] = "awesome"
x[:qualification] = x.delete(:qualifications)

3.1.0 :006 > x
 => {:qualification=>"awesome"}

Comment (goodviber) [reply app/services/find/match_old_params.rb:48]: So in the test 
```
context "with QUALIFICATION_FILTERS" do
      subject {
        described_class.call({
          "qualifications" => %w[Other PgdePgceWithQts QtsOnly],
        })
      }

      it "maps the old find params" do
        expect(subject).to eq({
          "qualification" => ["pgce pgde", "pgce_with_qts", "qts"],
```

File: app/views/find/result_filters/_hidden_fields.html.erb (hunk 4 above)
Comment (goodviber) [reply app/views/find/result_filters/_hidden_fields.html.erb]: Done

File: spec/services/find/match_old_params_spec.rb (hunk 6 above)
Comment (goodviber) [reply spec/services/find/match_old_params_spec.rb]: Done

File: app/services/find/match_old_params.rb (hunk 3 above)
Comment (goodviber) [reply app/services/find/match_old_params.rb:48]: Its 'qualificationS' mapped to 'qualification'

Comment (defong) [reply app/services/find/match_old_params.rb:48]: :woozy_face:

**Review by defong — APPROVED** [review]
